### PR TITLE
Use VTune task API with timers

### PR DESCRIPTION
**Pull request by markdewing**
Enable with -DUSE_VTUNE_TASKS=1.

Will need to add the VTune include directory to QMC_INCLUDE
and add the VTune lib directory (-L<vtune lib dir>) to CMAKE_CXX_FLAGS.

Also requires a knob or option to be enabled at collection time as the task API information is not collected by default.

This could also serve as a guide for adding annotations for other profiling tools.

### CODE REVIEW

**Comment by prckent** [comment]
Thanks. Please add a chapter to the manual "Using QMCPACK with external tools" with an entry for Intel VTune describing this, the vtune invocation, and any other relevant VTune usage. I will add Scitools Understand and ask @atillack / @yingwaili to put the ScoreP recipe. While these entries will be short it advertises that these tools have been made to work etc.

**Comment by ye-luo** [comment]
I think we need another flag, USE_VTUNE_API which is mainly to check the include and library of the API. We should check in CMake if the include path and library path are available.
QMC_INCLUDE should be avoid.
USE_VTUNE_TASKS will automatically trigger USE_VTUNE_API but users can also directly access it.

**Comment by ye-luo** [comment]
What is the prerequisites of USE_VTUNE_TASKS? ENABLE_TIMERS? We need to check these at CMake.

File: src/config.h.cmake.in [hunk 1]
```diff
@@ -209,5 +209,8 @@
 /* Internal timers */
 #cmakedefine ENABLE_TIMERS @ENABLE_TIMERS@
 
+/* Use VTune Task API with timers */
+#cmakedefine USE_VTUNE_TASKS @USE_VTUNE_TASKS@
```
Comment (ye-luo) [inline src/config.h.cmake.in]: Please add a macro for USE_VTUNE_API before the USE_VTUNE_TASKS

**Comment by markdewing** [comment]
I'm planning to close this PR and open an new one with the requested changes.

**Comment by markdewing** [comment]
Superseded by #642

**Pull request closed without merging**